pkg/innerring/processors/container/process_container.go

```diff
@@ -252,6 +255,147 @@ func (cp *Processor) approveDeleteContainer(e containerEvent.RemoveContainerRequ
 	}
 }
 
+func (cp *Processor) processSetAttributeRequest(req containerEvent.SetAttributeRequest) {
+	if !cp.alphabetState.IsAlphabet() {
+		cp.log.Info("non alphabet mode, ignore attribute setting")
+		return
+	}
+
+	id, err := cid.DecodeBytes(req.ID)
+	if err != nil {
+		cp.log.Error("attribute setting check failed",
+			zap.Error(fmt.Errorf("invalid container ID: %w", err)))
+		return
+	}
+	if id.IsZero() {
+		cp.log.Error("attribute setting check failed",
+			zap.Error(cid.ErrZero))
+		return
+	}
+
+	err = cp.checkSetAttributeRequest(req, id)
+	if err != nil {
+		cp.log.Error("attribute setting check failed",
+			zap.Stringer("container", id), zap.Error(err))
+		return
+	}
+
+	cp.approveSetAttributeRequest(req, id)
+}
+
+func (cp *Processor) checkSetAttributeRequest(req containerEvent.SetAttributeRequest, id cid.ID) error {
+	if now := time.Now().Unix(); now > req.ValidUntil {
+		return fmt.Errorf("request is valid until %d, now %d", req.ValidUntil, now)
```

Comment:
could be formatted to a human time version?

Reply:
added